src/openms/include/OpenMS/ANALYSIS/QUANTITATION/IsobaricQuantitationMethod.h

```diff
@@ -70,35 +71,35 @@ namespace OpenMS
       String description;
       /// The expected centroid position of the channel peak in m/z.
       Peak2D::CoordinateType center;
+      /// Ids of the affected channels. Order has to match the ones from the correction matrix parameter.
+      std::vector<Int> affected_channels;
 
       /// C'tor
-      IsobaricChannelInformation(const String local_name,
+      IsobaricChannelInformation(String local_name,
                                  const Int local_id,
-                                 const String& local_description,
+                                 String  local_description,
                                  const Peak2D::CoordinateType& local_center,
-                                 const Int minus_2,
-                                 const Int minus_1,
-                                 const Int plus_1,
-                                 const Int plus_2) :
-        name(local_name),
+                                 std::vector<Int>&& affected_channels) :
+        name(std::move(local_name)),
         id(local_id),
-        description(local_description),
+        description(std::move(local_description)),
         center(local_center),
-        channel_id_minus_2(minus_2),
-        channel_id_minus_1(minus_1),
-        channel_id_plus_1(plus_1),
-        channel_id_plus_2(plus_2)
+        affected_channels(affected_channels)
       {
       }
 
-      /// Id of the -2 isotopic channel (== -1 -> no channel)
-      Int channel_id_minus_2;
-      /// Id of the -1 isotopic channel (== -1 -> no channel)
-      Int channel_id_minus_1;
-      /// Id of the +1 isotopic channel (== -1 -> no channel)
-      Int channel_id_plus_1;
-      // Id of the +2 isotopic channel (== -1 -> no channel)
-      Int channel_id_plus_2;
+      IsobaricChannelInformation(String local_name,
+                                 const Int local_id,
```

Comment:
ditch the C'tors please and rely on aggregate inititialization (i.e. copy).
C'tors are called a handful of times in the whole program. Not worth just to save a few move's IMHO.

Reply:
I agree, I am not sure about pyopenms though ;)
Can we just keep it?

Reply:
we have two constructors. One for copy, one for move. I don't see the problem.

Reply:
It's four lines, 10 mins less work and one cycle of builds less for added flexibility.

Reply:
If the PR is of high quality, you need to live with the nitpicking ;)
fine as it is